### PR TITLE
fix(web): keep archived list pagination on unarchive

### PR DESCRIPTION
**Pull request by ColeMurray**
## Summary

- The Data Controls archive list optimistically hid rows on Unarchive but never rolled back `hiddenIds` on failure, and on success the `mutate(ARCHIVED_SESSIONS_KEY)` revalidation collapsed any "Load more" pagination via the SWR `onSuccess` callback.
- Switched to the request-first + scoped cache-update pattern already used by `session-sidebar.tsx` (`handleSessionArchived`): on success, remove the row from the first-page SWR cache with `revalidate: false` and filter it from `extraSessions`; on failure, leave state alone so the row stays visible.
- Adds a Vitest + RTL test file covering the affected behavior.

## Behavior change

| Scenario | Before | After |
|---|---|---|
| Unarchive succeeds | Row hidden optimistically, then revalidation collapses page-2 of "Load more" | Row removed from page-1 cache + extras filter; pagination preserved |
| Unarchive returns 500 | Row hidden, sometimes flickers back on revalidation, stays hidden if GET also fails | Row stays visible; `toast.error` shown |
| Unarchive succeeds on a page-2 session | Row reappears after revalidation collapse + onSuccess wipe | Row removed cleanly from `extraSessions` |
| Sidebar refresh on success | `mutate(SIDEBAR_SESSIONS_KEY)` (kept) | Same — newly active session reappears in sidebar |

## Out of scope (flagged for follow-up)

- The `onSuccess` callback on the archived-sessions `useSWR` still resets `extraSessions` on unrelated revalidations (focus/reconnect). The sidebar handles the same issue with a `prevDataRef` synchronous-render pattern (`session-sidebar.tsx:96-101`); worth porting in a separate PR.
- No tests added for the `/api/sessions/[id]/unarchive` route handler — would match the `route.test.ts` pattern from existing API route tests.

## Test plan

- [x] `npx vitest run` in `packages/web` — 30 files, 242 tests pass (5 new in `data-controls-settings.test.tsx`)
- [x] `npx tsc --noEmit` in `packages/web` — clean
- [x] `npx eslint src/` in `packages/web` — clean
- [ ] Manual: Settings > Data Controls — unarchive a session, confirm row disappears and reappears in sidebar
- [ ] Manual: Settings > Data Controls — force POST `/api/sessions/<id>/unarchive` to 500 via devtools, confirm row stays visible and an error toast appears
- [ ] Manual: Settings > Data Controls — with `>20` archived sessions, click "Load more", then unarchive a session from page 1; confirm page-2 rows remain visible

## Summary by CodeRabbit

* **Bug Fixes**
  * Session unarchive now shows clear success and error notifications and preserves pagination visibility when unarchiving across pages.
  * Sidebar reliably refreshes to reflect session unarchives.

* **Tests**
  * Added test coverage for unarchive success/failure, pagination edge cases, UI row visibility, and sidebar refresh behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/ColeMurray/background-agents/pull/635)

### CODE REVIEW

File: packages/web/src/components/settings/data-controls-settings.tsx [hunk 1]
```diff
@@ -51,21 +53,25 @@ export function DataControlsSettings() {
...
+      toast.success("Session unarchived");
+      await mutate<SessionListResponse>(
+        ARCHIVED_SESSIONS_KEY,
+        (current) =>
+          current
+            ? { ...current, sessions: removeSessionFromList(current.sessions, sessionId) }
+            : current,
+        { revalidate: false, populateCache: true }
+      );
+      setExtraSessions((prev) => prev.filter((s) => s.id !== sessionId));
```
[the remark this answers is not in the document]
Comment (ColeMurray) [reply packages/web/src/components/settings/data-controls-settings.tsx:71]: Fixed in 639ae7f0. Added `setOffset(prev => prev - 1)` on the success path and a regression test: 23 archived sessions, load page 1, unarchive session 1, click Load more, assert Session 21 appears. With the previous code Session 21 was skipped because the next fetch used the stale `offset=20` instead of `19`.

**Comment by ColeMurray** [comment]
CodeRabbit nit (SWRConfig duplication in the test file) addressed in 639ae7f0 — threaded an optional `additionalChildren` arg through `renderComponent` so the sidebar-invalidation test reuses the wrapper. Also refactored the fetch mock to a flat `archivedSessions` list, which models the server-side offset shift naturally.
